datahub-actions/src/datahub_actions/plugin/action/propagation/docs/propagation_action.py

```diff
@@ -0,0 +1,645 @@
+
+    def should_propagate(
+        self, event: EventEnvelope
+    ) -> Optional[DocPropagationDirective]:
+
+        if self.mcl_processor.is_mcl(event):
+            return self.mcl_processor.process(event)
+        if event.event_type == "EntityChangeEvent_v1":
+            assert isinstance(event.event, EntityChangeEvent)
+            # logger.info(f"Received event {event}")
+            assert self.ctx.graph is not None
+            semantic_event = event.event
+            if (
+                semantic_event.category == "DOCUMENTATION"
+                and self.config is not None
+                and self.config.enabled
+            ):
+                logger.debug("Processing EntityChangeEvent Documentation Change")
+                if self.config.columns_enabled and (
+                    semantic_event.entityType == "schemaField"
+                ):
+                    if semantic_event.parameters:
+                        parameters = semantic_event.parameters
+                    else:
+                        parameters = semantic_event._inner_dict.get(
+                            "__parameters_json", {}
+                        )
+                    doc_string = parameters.get("description")
+                    origin = parameters.get("origin")
+                    origin = origin or semantic_event.entityUrn
+                    via = (
+                        semantic_event.entityUrn
+                        if origin != semantic_event.entityUrn
+                        else None
+                    )
+                    logger.debug(f"Origin: {origin}")
+                    logger.debug(f"Via: {via}")
+                    logger.debug(f"Doc string: {doc_string}")
+                    logger.debug(f"Semantic event {semantic_event}")
+                    if doc_string:
+                        return DocPropagationDirective(
+                            propagate=True,
+                            doc_string=doc_string,
+                            operation=semantic_event.operation,
+                            entity=semantic_event.entityUrn,
+                            origin=origin,
+                            via=via,  # if origin is set, then via is the entity itself
+                            actor=(
+                                semantic_event.auditStamp.actor
+                                if semantic_event.auditStamp
+                                else self.actor_urn
+                            ),
+                        )
+        return None
+
+    def modify_docs_on_columns(
+        self,
+        graph: AcrylDataHubGraph,
+        operation: str,
+        schema_field_urn: str,
+        dataset_urn: str,
+        field_doc: Optional[str],
+        context: SourceDetails,
+    ) -> Optional[MetadataChangeProposalWrapper]:
+        if context.origin == schema_field_urn:
+            # No need to propagate to self
+            return None
+
+        if not dataset_urn.startswith("urn:li:dataset"):
```

Comment:
we have utils e.g. `from datahub.metadata.urns import Dataset` or `guess_entity_type` - this startswith check feels a bit too hardcoded

Reply:
i have seen this in a code quite a bit recently. i agree with avoiding raw strings for urn checks. seems like python sdk should have internal utilities for all of this